### PR TITLE
fix: raise exceptions in path importer

### PR DESCRIPTION
**Pull request by jemmyshin**
**Goals:**
Currently, we add all paths into `not_python_module_paths` but sometimes the exception comes from ImportError(or other types of error). So we need to raise this exception to make the error message clear.

- resolves clip-as-service issue [#865](https://github.com/jina-ai/clip-as-service/issues/865)

- [x] check and update documentation. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#-contributing-documentation) and ask the team.

### CODE REVIEW

File: tests/unit/test_importer.py [hunk 1]
```diff
@@ -48,3 +48,9 @@ def test_no_suppress_other_exception():
     with pytest.raises(Exception):
         with ImportExtensions(required=True, logger=default_logger):
             raise Exception
+
+
+def test_path_importer():
+    from jina.importer import PathImporter
+    with pytest.raises(ImportError):
+        PathImporter.add_modules('unit.error_package')
```
Comment (numb3r3) [inline tests/unit/test_importer.py:56]: The `ModuleNotFoundError` is not covered in this case.

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -370,7 +370,7 @@ jobs:
       - name: Test
         id: test
         run: |
-          pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml --timeout=600 -v -s --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
+          pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml --timeout=600 -v -s --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }} --ignore-glob=tests/unit/error_package.py
```
Comment (numb3r3) [inline .github/workflows/ci.yml:373]:
```suggestion
          pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml --timeout=600 -v -s --ignore-glob='tests/integration/hub_usage/dummyhub*'  --ignore=tests/unit/error_package.py ${{ matrix.test-path }}
```

File: tests/unit/test_importer.py (hunk 1 above)
Comment (jemmyshin) [reply tests/unit/test_importer.py:56]: this error won't be handled by this function

Comment (numb3r3) [reply tests/unit/test_importer.py:56]: try this

```
with pytest.raises(FileNotFoundError):
    PathImporter.add_modules('some_package.py')
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -370,7 +370,7 @@ jobs:
       - name: Test
         id: test
         run: |
-          pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml --timeout=600 -v -s --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
+          pytest --suppress-no-test-exit-code --force-flaky --min-passes 1 --max-runs 5 --cov=jina --cov-report=xml --timeout=600 -v -s --ignore=tests/unit/error_package.py  --ignore-glob='tests/integration/hub_usage/dummyhub*' ${{ matrix.test-path }}
```
Comment (numb3r3) [inline .github/workflows/ci.yml:373]: use `tmpfile` to create the dummy module instead in the unittest.

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (numb3r3) [reply .github/workflows/ci.yml:373]: don't change CI, it is too dangerous.

**Review by JoanFM — APPROVED** [review]

File: tests/conftest.py [hunk 4]
```diff
@@ -80,6 +80,14 @@ def tmpfile(tmpdir):
...
 
 
+@pytest.fixture()
+def tmpmodule(tmpdir):
+    tmpfile = f'package.py'
+    with open(tmpdir / tmpfile, 'w') as f:
+        f.write("raise ImportError")
+    return tmpdir / tmpfile
+
+
```
Comment (samsja) [inline tests/conftest.py:90]: if this fixture is only needed once lets move it inside the test or at least inside the test folder

**Review by numb3r3 — APPROVED** [review]
LGTM

**Review by samsja — APPROVED** [review]
lgtm

**Review by JoanFM — APPROVED** [review]

**Pull request merged**